### DISCUSSION TITLE
Bewegungsmelder ignorieren Einstellungen

### DISCUSSION THREAD

**Discussion by theimo1221** [OpenCCU/OpenCCU#1196] [🇩🇪 Generelle Diskussionen]
Meine Bewegungsmelder legen Zeiten regelmäßig scheinbar so aus wie sie wollen.

**Kontext:**
Ich nutze die Bewegungsmelder mittels ioBroker (rpc) in meiner eigenen HomeAutomation (NodeJS über Socket.io an ioBroker).
Falls Bewegungsmelder 10 Minuten keinen neuen Wert melden, interpretiere ich es lokal, als hätten sie sich selber zurückgesetzt.
Dies war eine Vorsichtsmaßnahme, nachdem sich sporadisch Bewegungsmelder selber nicht mehr gemeldet haben, dass nun die Bewegung wieder zu Ende ist.

Vermehrt tritt dieses Verhalten auf, wenn ein Bewegungsmelder vor Ablauf der Zurücksetzzeit erneut auslöst.

**Gegenbeispiel**
`I20210327-23:09:59.999(1)? Neuer Bewegunsstatus Wert für "Bewegungsmelder Treppenaufgang Oben": true`
...
`I20210327-23:14:58.601(1)? Neuer Bewegunsstatus Wert für "Bewegungsmelder Treppenaufgang Oben": false`

**Steps to reproduce the behavior**
1. Bewegungsmelder mit Einstellung `Mindestsendeabstand  = 2 Minuten; Zurücksetzen der Bewegung = 4 Minuten` betreiben
2. Eine Bewegung registrieren lassen
3. Differenzen der "Zurücksetz Zeit" festtellen

**Expected behavior**
Bewegungsmelder hält sich exakt an die vorgegebenen Zeiten

**Screenshots**
![image](https://user-images.githubusercontent.com/20605452/112736518-a2b8a700-8f53-11eb-9f8f-16567abc03da.png)

![image](https://user-images.githubusercontent.com/20605452/112736905-a1d54480-8f56-11eb-8c6c-cf95d8ad0581.png)

Im Abschnitt oben, sind die Zeiten generell spooky, da die Zweite Meldung nicht mind. 2 Minuten alt ist. 
Anschließend kommt erst nach 4,5 Minuten die erneute Meldung der erneuten Bewegung --> hier hätte ein "Keine Bewegung erkannt" nach 4 Minuten mit anschließendem "Neue Bewegung erkannt" erfolgen müssen

**System information:**
 - RaspberryMatic Version: 3.57.4.20210320
 - Used Hardware: RaspBerry Pi4
 - Used HomeMatic RF-Module: RF USB

**Comment by jens-maus** [comment]
Tut mir leider aber auf dieses Verhalten hat RaspberryMatic selbst keinerlei Einfluss, denn die Bewegungsmeldungen finden völlig autark im Bewegungsmelder selbst statt. RaspberryMatic kann hierbei nur die jeweiligen Parameter (Sendeabstand, Empfindlichkeit, etc.) im Bewegungsmelder einstellen. Die gesamte Bewegungserkennung und Meldelogik passiert im Gerät selber und darauf hat RaspberryMatic keinen Einfluss. D.h. es leitet die Eventmeldungen die vom Bewegungsmelder kommen einfach stumpf dann an die registrierten Schnittstellenprozesse (in deinem Fall ioBroker) weiter.

Und da die Firmware der Bewegungsmelder closed source ist kat RaspberryMatic selbst auf das Auslöseverhalten keinerlei Einfluss.

**Comment by theimo1221** [comment]
Hm....
Die RaspberryMatic, könnte allerdings selber einen Fallback auslösen, wenn das Gerät über die Rücksetzzeit hinaus nicht passend eine erneute Bewegung (oder seine Absenz) meldet, korrekt?

**Comment by jens-maus** [comment]
Inwiefern? Also zumindest wüsste ich nicht wie? Aber vielleicht hat ja @jp112sdl eine Idee dazu oder ich hab das Problem noch nicht ganz durchschaut. Wie gesagt findet die gesamte Auswertung der Bewegung komplett autark im Gerät selbst statt und RM hat darauf (ausser mittels umstellen der Parameter) keinerlei direkten Einfluss.

**Comment by theimo1221** [comment]
Unabhängig von dem falschen/autarken Verhalten des Gerätes selbst, wäre es m.E.n. nicht verkehrt, wenn die Raspberrymatic sagt: "Wenn ich nach einer gemeldeten Bewegung `Rücksetzzeit + x` nichts vom Gerät erhalte, dann gehe ich davon aus, dass die Bewegung aufgehört hat und setze den Wert in der Zentrale sicherheitshalber zurück auf false."

**Comment by jp112sdl** [comment]
> "Wenn ich nach einer gemeldeten Bewegung Rücksetzzeit + x nichts vom Gerät erhalte, dann gehe ich davon aus, dass die Bewegung aufgehört hat und setze den Wert in der Zentrale sicherheitshalber zurück auf false."

Das Problem ist meiner Meinung nach der Schnittstellenprozess. Dieser muss ja der ReGa das Event melden.
Heißt: Bei klassischen BidCos Geräten ist es der RFD, der die Events an die ReGa weiterreicht.
Bei HmIP ist es der HMIPServer Prozess.

So als Brainstorming Gedanke:

Du könntest versuchen, es mit einem Skript zu lösen und das Event faken.
Ich hatte da mal vor langer Zeit von Alchy ein Snippet erhalten, um eine ausgelöste "Batterie leer" Meldung wieder loszuwerden. 
```
string cmd = "/bin/sh -c '" # 'echo "load tclrpc.so; puts [xmlrpc http://127.0.0.1:1999/ event [list string BidCos-RF] [list string "'#sserial#'"] [list string LOWBAT] [list bool "'#bcmd#'"]];" |tclsh'# "'";
```
_Ich weiß aber nicht, ob es reicht, einfach aus BidCos-RF ein HmIP-RF zu machen._

Dann prüfst du zyklisch mit `.LastTimeStamp()` wann sich das Gerät das letzte Mal gemeldet hat und führst das Event-Skript ggf. aus.

**Comment by jens-maus** [comment]
@theimo1221 Ich hoffe es ist ok das ich dein Issue jetzt in einen Diskussionsbeitrag umgewandelt habe, denn ich sehe hier weiterhin keinen wirklichen Ansatz hier inhaltlich in RaspberryMatic selbst etwas daran zu ändern. Wie gesagt funktioniert diese ganze Bewegungsmeldung/auswertung völlig autark im Gerät und die Bewegungsevents werden lediglich an RaspberryMatic gemeldet ohne das da eine Auswertung passiert. Und wenn dann müsste man das wie @jp112sdl eben im Nutzerkontext irgendwie umsetzen wenn man das verhalten anders haben will - oder eben das als Verbesserungsvorschlag an eQ3 melden.

**Comment by Baxxy13** [comment]
Für mich ist das Eindeutig ein Fehler in der Firmware des Bewegungsmelders!
Testgerät hier : HmIP-SMI55 (FW: 1.0.12)
Mit den oben gezeigten Einstellungen konnte ich das Verhalten bei 3/3 Versuchen jedes mal reproduzieren.
Wie vermutet tritt das Problem auf wenn nach Ablauf der 2 Minuten Mindestsendeabstand eine erneute Bewegung registriert wird. Dann bleibt der Status bei "Bewegung erkannt" hängen.

Das es bei Dir nicht immer auftritt (sowie die "zwischengeschobenen" Meldungen) könnte mit den zykl. Statusmeldungen, oder mit den Sendungen bei Änderung des Helligkeitswertes zusammenhängen. Das kann man aber effektiv nur in einer Dunkelkammer testen.

Edit: Nach rund 45 Minuten hat mein SMI55 eine zykl. Meldung gesendet und damit auch wieder den Status auf "keine Bewegung erkannt" gesetzt.

Theoretisch könntest du also dein Problem umgehen indem du den Intervall der zykl. Meldungen verringerst. Was aber dann sicher auf die Batterielaufzeit und eventuell den Duty Cycle Auswirkungen hat.

Letztlich kann aber nur eQ-3 das mit einem Firwareupdate wirklich fixen.

So, lange genug stillgesessen... ;-)

**Comment by blackbasket** [comment]
Das Problem mit den Bewegungsmeldern ist m. E. deutlich umfangreicher. Immerhin habe ich eQ3 nach mehreren Wochen davon überzeugen können, dass bei HmIP-SMO der Ch2 gar nicht gesteuert werden kann. Selbst in DVs.

Das war ja leider schon schwer genug, da es erst wochenlang an CUxD gelegen haben soll^^

**Reply by Baxxy13** [reply]
Naja, wenn ich ein's weiß, dann das CUxD immer Schuld ist! ;-)



> HmIP-SMO der Ch2 gar nicht gesteuert werden kann. Selbst in DVs.

Das ist "Unterdrückung der Bewegungserkennung", oder? Der Kanal lässt sich doch eh nur mit DV verwenden und nicht in Programmen. Und DV geht da beim SMO auch nicht?
Da habe ich mit meinem SMI55 noch Glück, den kann ich per DV "ausknipsen". Aber leider nur maximal 31 Minuten obwohl die WebUI (und auch die techn. Dokumentation) maximal 31h vorsieht. Da scheint die Firmware intern die Zeitparameter falsch zu interpretieren. eQ-3 melden? ---> sie haben CUxD installiert.

**Reply by blackbasket** [reply]
Genau so... das ist auch einfach nur peinlich... ich würde mich schämen, da im "Support" zu arbeiten...

**Comment by theimo1221** [comment]
22:58:55.959 Presence Update (true) for "1. OG Kueche Praesenzmelder"
22:59:24.286 Presence Update (true) for "1. OG Kueche Praesenzmelder"
23:01:01.795 Presence Update (false) for "1. OG Kueche Praesenzmelder"
23:01:01.796 Bewegung im Raum Kueche zu Ende --> Licht ausschalten
23:01:02.174 Presence Update (true) for "1. OG Kueche Praesenzmelder"
23:01:02.175 Bewegung im Raum Kueche festgestellt --> Licht einschalten
23:01:36.943 Presence Update (true) for "1. OG Kueche Praesenzmelder"
…
23:03:07.901 Presence Update (false) for "1. OG Kueche Praesenzmelder"
23:03:07.902 Bewegung im Raum Kueche zu Ende --> Licht ausschalten


Ist übrigens beim Präsenzmelder auch fehlerhaft aber anders.

Einstellung in dem Raum 

- Reset nach 2 Minuten
- Erneut senden nach 30s
- Innerhalb des Sendeabstands nicht neu senden

Sprich er hätte das erste Mal frühestens um 23:01:24 auf false setzen dürfen und beim 2. Mal ist er auch 30s zu früh.

Was man sagen kann ist, dass der Präsenzmelder seinen Timer nicht neustartet (bei erkannter Bewegung) aber zumindest seinen Timer seit der ersten erkannten Bewegung einhält.
